brave/src/test/java/brave/propagation/ExtraFieldPropagationTest.java

```diff
@@ -105,12 +110,58 @@
     TraceContext context = extractWithAmazonTraceId();
 
     try (Tracing t = Tracing.newBuilder().propagationFactory(factory).build();
-         CurrentTraceContext.Scope scope = t.currentTraceContext().newScope(context)) {
+         Scope scope = t.currentTraceContext().newScope(context)) {
       assertThat(ExtraFieldPropagation.get("x-amzn-trace-id"))
         .isEqualTo(awsTraceId);
     }
   }
 
+  @Test public void emptyFields_disallowed() {
+    assertThatThrownBy(() -> ExtraFieldPropagation.newFactory(B3Propagation.FACTORY, ""))
+      .hasMessage("fieldNames[0] is empty");
+
+    assertThatThrownBy(() -> ExtraFieldPropagation.newFactory(B3Propagation.FACTORY, asList("")))
+      .hasMessage("fieldNames[0] is empty");
+
+    assertThatThrownBy(() -> newFactoryBuilder(B3Propagation.FACTORY).addField("").build())
+      .hasMessage("fieldName is empty");
+
+    assertThatThrownBy(() -> newFactoryBuilder(B3Propagation.FACTORY).addRedactedField("").build())
+      .hasMessage("fieldName is empty");
+
+    assertThatThrownBy(
+      () -> newFactoryBuilder(B3Propagation.FACTORY).addPrefixedFields("foo", asList("")).build())
+      .hasMessage("fieldNames[0] is empty");
+  }
+
+  // We formerly enforced presence of field names in the factory's factory method
+  @Test public void noFields_newFactory_disallowed() {
+    assertThatThrownBy(() -> ExtraFieldPropagation.newFactory(B3Propagation.FACTORY))
+      .hasMessage("no field names");
+
+    assertThatThrownBy(() -> ExtraFieldPropagation.newFactory(B3Propagation.FACTORY, asList()))
+      .hasMessage("no field names");
+  }
+
+  // We formerly accepted .build() when no fields were present
```

Comment:
@meltsufin  this bud's for you

regardless of if extra fields are added to the builder or not, behavior will be predictable

Reply:
Thanks! Much appreciated! 😃